### DISCUSSION TITLE
interval analog input update

### DISCUSSION THREAD

**Discussion by tsunfinger** [arendst/Tasmota#19668] [Q&A]
is there an option to change update interval of analog input? On changes it looks like I only get updates about once per second by
**on Analog#A1**
and value is an average of intervall

**Comment by sfromis** [comment]
No such option. With ESP32, you could try using Berry code to ask for values at a faster cycle, something like this:
```
class sensorhammer
  def init()
    tasmota.add_driver(self)
    tasmota.set_timer(60000,/->tasmota.remove_driver(self))
  end
  def every_250ms()
    tasmota.set_timer(0,/->print(tasmota.read_sensors(true)))
  end
end
sensorhammer()
```
Beware that the Berry console will not show all the messages, better see results in the Tasmota console.

**Comment by pkkrusty** [comment]
With scripting, you will get updates much faster and choose your averaging. Available on 8266.

**Reply by tsunfinger** [reply]
But this isn't included in precompiled images!?

**Reply by Jason2866** [reply]
What's your use case? Better describe what you want to solve than digging in details which maybe are not needed.

**Reply by tsunfinger** [reply]
I bought a chicken door but original software is ... not usable. Hardware is based on ESP32 with H-bridge for door engine. I have it running on Tasmota with rules. The challenge at all - the hardware has no end-switch. It means I don't know when door is closed/opened. I controlling H-bridge time based, so I switch it on for about 45 seconds (PWM controlled).
Now I have decided to use analog input (used on hardware for measuring external power/voltage) to indicate door is opened/closed. If engine is blocked the power consumption will increase and voltage on analog input decreases.
It works not perfectly because the analog input delivers updates on rules every 5 seconds (?) but for my use case it must be more often (at least every 500ms).

**Reply by Jason2866** [reply]
Use Berry, as sfromis already suggested.

**Reply by barbudor** [reply]
Isn't that a use-case for Tasmota Shutter feature ?

**Reply by Jason2866** [reply]
Yes, shutter should work.

**Reply by tsunfinger** [reply]
Shutter on a generic DC motor? Following the example on shutter it expect a stepper/servo motor.

**Reply by barbudor** [reply]
read better the whole page
